test/src/test/java/hudson/console/HyperlinkNoteTest.java

```diff
@@ -72,6 +80,28 @@ public void textWithNewlinesModelHyperlinkNote() throws Exception {
                 containsString(new ModelHyperlinkNote("", 0).extraAttributes()),
                 containsString(">" + noteTextSanitized + "</a>")));
     }
+    
+    @Test
+    public void textWithSingleQuote() throws Exception {
+    	FreeStyleProject upstream = r.createFreeStyleProject("upstream");
+    	r.createFreeStyleProject("d0wnstr3'am");
+    	upstream.getPublishersList().add(new BuildTrigger("d0wnstr3'am", Result.SUCCESS));
+    	FreeStyleBuild b = r.buildAndAssertSuccess(upstream);
+        r.waitUntilNoActivity();
+        HtmlPage rsp = r.createWebClient().goTo(b.getUrl()+"console");
+        System.out.println("Before:\n"+rsp.getWebResponse().getContentAsString()+"\n\n");
```

Comment:
Debugging println statements are better expressed as assertions of an expectation or can be removed if their debugging purpose has been completed.
```suggestion
```